### ISSUE TITLE
Template variable not updating correctly with graphite query

### ISSUE THREAD

**Issue by snc29590** [grafana/grafana#5103]
Thank you for helping us make Grafana even better!

To help us respond to your issues more quickly, please make sure to add as much information as possible.

If this issue is about a plugin, please open the issue in that plugin's repository.

Start your issue's title with [Feature Request] / [Bug] / [Question] or no tag if you're unsure. Also, please be aware that GitHub now supports uploading of screenshots; look at the bottom of this input field.

Please include some basic information:
- What Grafana version are you using?
- What datasource are you using?
- What OS are you running grafana on?
- What did you do?
- What was the expected result?
- What happenend instead?

If your question/bug relates to a metric query / unexpected data visualization, please include:
- An image or text representation of your metric query
- The raw query and response from your data source (check this in chrome dev tools network tab)

**Comment by snc29590** [comment]
We are using grafana-3.0.2-1463383025.x86_64 on a Centos 6 machine with graphite datasource.

We have a graphite setup where the metrics are stored with the namespace:
`metrics.<env>.<group>-<type>.<instance>.`

Example: `metrics.prod.grp1-mysql.01.`

The grafana dashboard has the following template variables:

variable: `env`
query: `metrics.*`

variable: `group`
query: `metrics.$env.*`
regex:`/([^-]*)-/`

variable:`type`
query: `metrics.$env.$group-*`
regex: `/-(.*)$/`

variable: `instance`
query: `metrics.$env.$group-$type.*`

Each "$group" has different server "types", hence, changing the group should also change the values for the "$type".

However, the problem I have is, when I change the "$group" variable, the "$type" variable does not get updated automatically. 

This used to work prefectly well on Grafana 2.6, but has stopped working since Grafana 3.0.

**Comment by torkelo** [comment]
It should work if you have multi-value disabled. 

The multi value option will not automatically change it's value (but the options will)
https://github.com/grafana/grafana/issues/2342


http://play.grafana.org/dashboard/db/templated-graphs-nested

**Comment by snc29590** [comment]
The multi-value function was never enabled on this dashboard, and still, it does not work. Here are some screenshots to show what is happening:

The following six metrics are present in graphite:
```metrics.prod.grp1-mysql.01.status
metrics.prod.grp1-vertica.01.status
metrics.prod.grp1-jazz.01.status
metrics.prod.grp2-oracle.01.status
metrics.prod.grp2-ui.01.status
metrics.prod.grp2-zabbix.01.status
```

On Grafana 2.6, when I select the "group" from the drop down, the "type" changes as per the "group" selection.
![image](https://cloud.githubusercontent.com/assets/16399189/15396139/3d698586-1dca-11e6-96f5-27b112910ce1.png)


The same dashboard, in 3.0 does not update the "type" when "group" is changed:
![image](https://cloud.githubusercontent.com/assets/16399189/15396183/69e912de-1dca-11e6-8c12-4df601364354.png)

**Comment by torkelo** [comment]
strange, cant replicate it. Can you show the network requests & responses when you change group?

**Comment by snc29590** [comment]
Here's the network request/response from 2.6 and 3.0.

In 2.6, when I change the "group" template variable from grp1 to grp2, it sends a request to graphite's "/query" end point and updates the subsequent variables.

![network-2 6](https://cloud.githubusercontent.com/assets/16399189/15401831/d19d5a3c-1de0-11e6-86b5-b7d0119454da.png)

When I do the same in 3.0, it does not send a request to graphite's "/query" end point at all. It simply hits the "/render" api to try and get metrics for non-existent metrics.

![network-3 0](https://cloud.githubusercontent.com/assets/16399189/15401841/d9ae0410-1de0-11e6-9a13-13de0ff0047d.png)

**Comment by torkelo** [comment]
very strange, the same setup works here
http://play.grafana.org/dashboard/db/templated-graphs-nested

**Comment by torkelo** [comment]
can try changing the type variable to this:

`metrics.$env.[[group]]-*`

**Comment by snc29590** [comment]
Nope, that does not help either. I get the same result. Could this be because of the fact that there are two template variables in the same node ($group-$type)?

My other dashboards that are for a different graphite datasource under a different namespace (one variable per node) work fine. It is only dashboards for this particular graphite datasource that stopped working after the move from 2.6 to 3.0.

**Comment by torkelo** [comment]
found the cause! fixed in 3.0.x branch will be included in next patch release (released soon, and in nightly 3.1 builds)

**Comment by snc29590** [comment]
cool, thank you very much for the follow up and the fix! :+1:
